### PR TITLE
initial contribution of a generic mDNS discovery approach

### PR DESCRIPTION
**Pull request by peuter**
now hopefully complete and signed correctly

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Thanks! Will try to review and test it in the next two weeks - thanks for your patience!

File: bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java [hunk 1]
```diff
@@ -8,81 +8,124 @@
...
-				0, 0, description.serviceProperties);
-		logger.debug("Unregistering service " + description.serviceType + " at port " + 
-				String.valueOf(description.servicePort));
-		jmdns.unregisterService(serviceInfo);
+		ServiceInfo serviceInfo = ServiceInfo.create(description.serviceType,
+				description.serviceName, description.servicePort, 0, 0,
+				description.serviceProperties);
+		logger.debug("Unregistering service " + description.serviceType
+				+ " at port " + String.valueOf(description.servicePort));
+		mdnsClient.getClient().unregisterService(serviceInfo);
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java]: shouldn't here be better a null check on mdnsClient?

File: bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java [hunk 2]
```diff
@@ -8,81 +8,124 @@
...
 	}
 
 	/**
 	 * This method unregisters all services from Bonjour/MDNS
 	 */
 	protected void unregisterAllServices() {
-		jmdns.unregisterAllServices();
+		mdnsClient.getClient().unregisterAllServices();
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java]: shouldn't here be better a null check on mdnsClient?

File: bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/discovery/MDNSDiscoveryService.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		this.mdnsClient = null;
+	}
+	
+	protected void activate() {
+		
+	}
+
+	@Override
+	protected void startScan() {
+		logger.debug("mDNS discovery service startet");
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/discovery/MDNSDiscoveryService.java]: "started" :-)

File: bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java [hunk 4]
```diff
@@ -8,81 +8,124 @@
...
-		}
+
 	}
-	
+
 	public void deactivate() {
 		unregisterAllServices();
 		try {
-			jmdns.close();
+			mdnsClient.getClient().close();
```
Comment (kaikreuzer) [inline bundles/io/org.eclipse.smarthome.io.mdns/src/main/java/org/eclipse/smarthome/io/mdns/internal/MDNSServiceImpl.java]: shouldn't here be better a null check on mdnsClient?

**Comment by kaikreuzer** [comment]
Thanks Tobias, very nice code!
Could you please run a "mvn license:format" over the code so that the license headers are updated for 2015?
I wonder, if with this contribution we should actually rename the bundle to io.transport.mdns - so that it is in sync with the upnp bundle. If you agree, it would be great if you could take care of this as well!

**Comment by peuter** [comment]
Ok I hope I got everything now:
- for the updating of the license headers I had to change the year property in the root pom.xml to 2015 because it is still 2014 there. I guess this should be commited also, but I didn´t want to mix up in this PR
- I renamed the bundle to io.transport.mdns and hopefully did not forget anything, as I almost totally messed up my clone with this I am not sure if everything is right now.
- I have no idea why some of my commits are not valid although the signed-off-by is added correctly, like the last one. Is there anything I can do there?

**Comment by kaikreuzer** [comment]
Cool, you are really quick, thanks!
Just one place where the bundle name still needs to be replaced: https://github.com/eclipse/smarthome/blob/master/features/org.eclipse.smarthome.feature.runtime.core/feature.xml#L121
The last thing I have to ask you for is to condense all your commits of this PR into a single (signed) commit. I hope this won't screw up your local repo again - better make a backup before :-)
See here: http://stackoverflow.com/a/14535190

**Comment by peuter** [comment]
Ok, done!

**Comment by kaikreuzer** [comment]
Thanks - looks almost good, but I now see two commits in this PR and the second one contains a lot of stuff, which is not really related: https://github.com/peuter/smarthome/commit/e8f254199cbf0a260a4d68ab41f23a8f2d7a5004
Any chance to fix that?

**Comment by peuter** [comment]
I think I will backup, refork and copy+paste to get a clean start. Should be easier than anything else.

**Comment by peuter** [comment]
Done in #182

**Pull request closed without merging**